example/hello_ll.c

```diff
@@ -242,11 +263,19 @@ int main(int argc, char *argv[])
 		goto err_out1;
 	}
 
+	if (fuse_opt_parse(&args, &lo, hello_ll_opts, NULL) == -1)
+		return 1;
+
 	se = fuse_session_new(&args, &hello_ll_oper,
 			      sizeof(hello_ll_oper), NULL);
 	if (se == NULL)
 	    goto err_out1;
 
+	if (lo.max_write) {
+		conn = fuse_get_conn_info(se);
+		conn->max_write = lo.max_write;
+	}
```

Comment:
Thank you, the intent is clear now, though now I don't understand why can't do the same in the ->init() method, which actually gets passed conn

In `do_init()`
```
	if (se->op.init)
		se->op.init(se->userdata, &se->conn);
```

Reply:
Yes, we can do it in ->init() method, but as i said before, there is a bug there:
```
diff --git a/example/hello_ll.c b/example/hello_ll.c
index bea2112..007d4f8 100644
--- a/example/hello_ll.c
+++ b/example/hello_ll.c
@@ -201,8 +201,15 @@ static void hello_ll_removexattr(fuse_req_t req, fuse_ino_t ino, const char *nam
                fuse_reply_err(req, ENOTSUP);
        }
 }
+static void hello_ll_init(void *userdata, struct fuse_conn_info *conn)
+{
+       (void)userdata;
+       conn->max_write = 512 * 4096;
+       return;
+}
 
 static const struct fuse_lowlevel_ops hello_ll_oper = {
+       .init = hello_ll_init,
        .lookup = hello_ll_lookup,
        .getattr = hello_ll_getattr,
        .readdir = hello_ll_readdir,
diff --git a/lib/fuse_lowlevel.c b/lib/fuse_lowlevel.c
index d9844b0..3f13f2a 100644
--- a/lib/fuse_lowlevel.c
+++ b/lib/fuse_lowlevel.c
@@ -2189,6 +2189,7 @@ void do_init(fuse_req_t req, fuse_ino_t nodeid, const void *inarg)
                fuse_log(FUSE_LOG_DEBUG, "   max_readahead=0x%08x\n",
                        outarg.max_readahead);
                fuse_log(FUSE_LOG_DEBUG, "   max_write=0x%08x\n", outarg.max_write);
+               fuse_log(FUSE_LOG_DEBUG, "   fuse_session_size=0x%08x\n", se->bufsize);
                fuse_log(FUSE_LOG_DEBUG, "   max_background=%i\n",
                        outarg.max_background);
                fuse_log(FUSE_LOG_DEBUG, "   congestion_threshold=%i\n",
```
I tried above codes, it'll fail:
[lege@fedora build]$ sudo ./example/hello_ll -d /home/lege/mntpoint/ 
FUSE library version: 3.16.2
unique: 2, opcode: INIT (26), nodeid: 0, insize: 104, pid: 0
INIT: 7.33
flags=0x53fffffb
max_readahead=0x00020000
   INIT: 7.39
   flags=0x40409029
   max_readahead=0x00020000
   max_write=0x00200000
   fuse_session_size=0x00101000
   max_background=0
   congestion_threshold=0
   time_gran=1
   unique: 2, success, outsize: 80
fuse: reading device: Invalid argument

The reason is that "conn->max_write + FUSE_BUFFER_HEADER_SIZE" does not match se->bufsize. This bug could be fixed by below diff:
```
diff --git a/lib/fuse_lowlevel.c b/lib/fuse_lowlevel.c
index d9844b0..99c6075 100644
--- a/lib/fuse_lowlevel.c
+++ b/lib/fuse_lowlevel.c
@@ -2079,13 +2079,13 @@ void do_init(fuse_req_t req, fuse_ino_t nodeid, const void *inarg)
        }
        se->bufsize = bufsize;
 
-       if (se->conn.max_write > bufsize - FUSE_BUFFER_HEADER_SIZE)
-               se->conn.max_write = bufsize - FUSE_BUFFER_HEADER_SIZE;
-
        se->got_init = 1;
        if (se->op.init)
                se->op.init(se->userdata, &se->conn);
 
+       if (se->conn.max_write > bufsize - FUSE_BUFFER_HEADER_SIZE)
+               se->conn.max_write = bufsize - FUSE_BUFFER_HEADER_SIZE;
+
```
then
[lege@fedora build]$ sudo ./example/hello_ll -d /home/lege/mntpoint/ 
FUSE library version: 3.16.2
unique: 2, opcode: INIT (26), nodeid: 0, insize: 104, pid: 0
INIT: 7.33
flags=0x53fffffb
max_readahead=0x00020000
   INIT: 7.39
   flags=0x40409029
   max_readahead=0x00020000
   max_write=0x00100000
   fuse_session_size=0x00101000
   max_background=0
   congestion_threshold=0
   time_gran=1
   unique: 2, success, outsize: 80

No error occurs, and it seems that libfuse will limit conn->max_write to not greater than se->bufsize, in fuse_session_new(), it sets "se->bufsize = FUSE_MAX_MAX_PAGES * getpagesize() + FUSE_BUFFER_HEADER_SIZE" initially.

So far, I only care about whether conn->max_write can be set correctly, so I'm ok to close this pr(not mergerd), but
above do_init()'s bug needs to be fixed, I can submit a patch, what do you think?
And make conn->max_write not greater than se->bufsize is libfuse's design intent? Can users set conn->max_write to be greater than the default " FUSE_MAX_MAX_PAGES * getpagesize()"?  whih may need explict documents, thanks.